### PR TITLE
bootchooser: put magic values in defines

### PR DESCRIPTION
**Pull request by jremmet**
Signed-off-by: Jan Remmet <j.remmet@phytec.de>

### CODE REVIEW

**Comment by jremmet** [comment]
Maybe storing this values in system.conf is an option? We could also fall-back to the now hard coded.

File: src/bootchooser.c [hunk 1]
```diff
@@ -152,16 +155,16 @@ static gboolean barebox_set_primary(RaucSlot *slot) {
...
-			prio = prio2;
+			prio = BAREBOX_STATE_DEFAULT_PRIORITY;
 		}
 		g_ptr_array_add(pairs, g_strdup_printf("%s.%s.priority=%i",
 				BOOTSTATE_PREFIX, s->bootname, prio));
 	}
 
 	g_ptr_array_add(pairs, g_strdup_printf("%s.%s.remaining_attempts=%i",
-			BOOTSTATE_PREFIX, slot->bootname, 3));
+			BOOTSTATE_PREFIX, slot->bootname, BAREBOX_STATE_NEW_ATTEMPS));
```
Comment (ejoerns) [inline src/bootchooser.c]: This should be `BAREBOX_STATE_DEFAULT_ATTEMPS`, too. There is no difference in meaning here.

File: src/bootchooser.c [hunk 2]
```diff
@@ -152,16 +155,16 @@ static gboolean barebox_set_primary(RaucSlot *slot) {
 			continue;
 
 		if (s == slot) {
-			prio = prio1;
+			prio = BAREBOX_STATE_NEW_PRIORITY;
```
Comment (ejoerns) [inline src/bootchooser.c]: I would prefer something like `BAREBOX_STATE_PRIORITY_PRIMARY` as it better reflects the purpose of this value.

File: src/bootchooser.c (hunk 1 above)
Comment (jremmet) [inline src/bootchooser.c]: I have a case were the customer want's different behaviour after an update. After Update there should be less attempts than later if the update works fine.
rauc mark-good will use BAREBOX_STATE_DEFAULT_ATTEMPS

File: src/bootchooser.c (hunk 2 above)
Comment (jremmet) [inline src/bootchooser.c]: changed this

**Pull request merged**